optional_plugins/varianter_pict/avocado_varianter_pict/__init__.py

```diff
@@ -0,0 +1,176 @@
+    for line in lines[1:]:
+        variants.append(dict(zip(headers, line.split('\t'))))
+    return (headers, variants)
+
+
+class VarianterPict(Varianter):
+
+    """
+    Processes the pict file into variants
+    """
+
+    name = 'yaml_to_mux'
+    description = 'Multiplexer plugin to parse yaml files to params'
+
+    def initialize(self, args):
+        self.variants = None
+        error = False
+
+        pict_parameter_file = getattr(args, "pict_parameter_file", None)
+        if pict_parameter_file is None:
+            return
+        if not os.access(pict_parameter_file, os.R_OK):
+            LOG_UI.error("pict parameter file '%s' could not be found or is "
+                         "not readable", pict_parameter_file)
+            error = True
+
+        pict_binary = getattr(args, "pict_binary", None)
+        if pict_binary is None:
+            LOG_UI.error("pict binary '%s' could not be found, please set it "
+                         "with --pict-binary or put it in your PATH",
+                         pict_binary)
+            error = True
+        elif not os.access(pict_binary, os.R_OK | os.X_OK):
+            LOG_UI.error("pict binary '%s' can not be executed, please check "
+                         "the option given with --pict-binary and/or the file "
+                         "permissions")
+            error = True
+
+        if error:
+            if args.subcommand == 'run':
+                sys.exit(exit_codes.AVOCADO_JOB_FAIL)
+            else:
+                sys.exit(exit_codes.AVOCADO_FAIL)
+
+        self.parameter_path = getattr(args, "pict_parameter_path")
+
+        output = run_pict(pict_binary,
+                          pict_parameter_file,
+                          getattr(args, "pict_order_of_combinations"))
+        self.headers, self.variants = parse_pict_output(output)
+
+    def __iter__(self):
+        if self.variants is None:
+            return
+
+        variant_ids = []
+        for variant in self.variants:
+            base_id = "-".join([variant.get(key) for key in self.headers])
+            variant_ids.append(base_id + '-' +
+                               hashlib.sha1(base_id).hexdigest()[:4])
+
+        for vid, variant in itertools.izip(variant_ids, self.variants):
+            variant_tree_nodes = []
+            for key, val in variant.items():
+                variant_tree_nodes.append(TreeNode(key, {key: val}))
+
+            yield {"variant_id": vid,
+                   "variant": variant_tree_nodes,
+                   "mux_path": self.parameter_path}
+
+    def __len__(self):
+        return sum(1 for _ in self)
+
+    def update_defaults(self, defaults):
+        pass
+
+    def to_str(self, summary, variants, **kwargs):
+        """
+        Return human readable representation
+
+        The summary/variants accepts verbosity where 0 means silent and
+        maximum is up to the plugin.
+
+        :param summary: How verbose summary to output (int)
```

Comment:
I really miss the `summary` implementation. I'd accept a Trello card in exchange :)

Reply:
https://trello.com/c/Aio6IvOs/1171-implement-different-summary-levels-for-pict-varianter